core/integration/suite_test.go

```diff
@@ -159,6 +159,39 @@ func startRegistry(ctx context.Context, c *dagger.Client, t *testing.T) {
 	require.NoError(t, err)
 }
 
+func startPrivateRegistry(ctx context.Context, c *dagger.Client, t *testing.T) {
+	t.Helper()
+
+	// Include a random ID so it runs every time (hack until we have no-cache or equivalent support)
+	creds := c.Host().Directory("testdata/auth")
```

Comment:
Any reason not to do this with a memory-backed file instead?

```go
c.Directory().WithNewFile("htpasswd", "john:$2y$05$/iP8ud0Fs8o3NLlElyfVVOp6LesJl3oRLYoc3neArZKWX10OhynSC")
```

Could perhaps generate this content in code from username/password consts or arguments - I was initially confused why I didn't see the password anywhere else, but it's because it's in htpasswd format here. It was also kind of scary since it looked like a real password, maybe it could be a dumb string like `test-password`? But that's really not a big deal.

Also the comment above is detached from its code below (`randomID := ...`)

Reply:
Nice catch! Fixed in https://github.com/dagger/dagger/pull/4502/commits/ba9c1ceed14ef618032ecc7d2bbec1584547452d